### ISSUE TITLE
Resolution of strong name keys without full paths doesn't work with OOP

### ISSUE THREAD

**Issue by KevinH-MS** [dotnet/roslyn#16563]
If you have an AssemblyKeyFileAttribute in code, and the argument is a relative path, it will not be resolved correctly if you try to compile in the OOP process (Compilation.Emit will generate Diagnostic saying they key file can't be resolved).

In AbstractOptionsSerializationService.cs:
```
160     strongNameProvider = new DesktopStrongNameProvider();
```
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Execution/AbstractOptionsSerializationService.cs,160

We need to be passing in the project file directory and the output directory for the project that the Compilation is derived from.

I also notice that there appears to be a bug in MSBuildProjectLoader.cs:
```
427     .WithStrongNameProvider(new DesktopStrongNameProvider(ImmutableArray.Create(projectDirectory, outputFilePath)))
```
should be:
```
427     .WithStrongNameProvider(new DesktopStrongNameProvider(ImmutableArray.Create(projectDirectory, outputDirectory)))
```

**Comment by KevinH-MS** [comment]
FYI @heejaechang

**Comment by srivatsn** [comment]
Fixed with #16655
